include/mbedtls/legacy_or_psa.h

```diff
@@ -209,4 +209,88 @@
 #define MBEDTLS_HAS_ALG_SHA_512_VIA_MD_OR_PSA_BASED_ON_USE_PSA
 #endif
 
+/* Ciphers and cipher chaining using low-level or PSA based on availability */
+#if defined(MBEDTLS_AES_C) || \
+    ( defined(MBEDTLS_PSA_CRYPTO_C) && defined(PSA_WANT_KEY_TYPE_AES) )
+#define MBEDTLS_HAS_ALG_AES_VIA_LOWLEVEL_OR_PSA
```

Comment:
As discussed in chat, this is not a good interface because it's expressed in terms of how it's implemented, rather than in terms of how to use it.

It's not yet clear to me how this is meant to be used, or whether this is necessary at all.